dev/ProgressRing/ProgressRing.cpp

```diff
@@ -41,18 +47,98 @@ void ProgressRing::OnRangeBasePropertyChanged(const winrt::DependencyObject&, co
     UpdateSegment();
 }
 
+void ProgressRing::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
+void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
 void ProgressRing::OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    ApplyLottieAnimation();
     UpdateRing();
 }
 
-winrt::Size ProgressRing::ComputeEllipseSize(double thickness, double actualWidth, double actualHeight)
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressRing::ApplyLottieAnimation()
+{
+    if (auto&& player = m_player.get())
+    {
+        // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
+        const float diameter = static_cast<float>(ActualWidth());
+        const auto size = winrt::Size({ diameter, diameter });
+
+        const auto [foregroundColor, backgroundColor] = [
+            foreground = Foreground().try_as<winrt::SolidColorBrush>(),
+            background = Background().try_as<winrt::SolidColorBrush>()]()
+        {
+            winrt::Color foregroundColor;
+            winrt::Color backgroundColor;
+
+            if (foreground)
+            {
+                foregroundColor = foreground.Color();
+            }
+            else
+            {
+                foregroundColor = SharedHelpers::FindResource(s_DefaultForegroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
+            }
+
+            if (background)
+            {
+                backgroundColor = background.Color();
+            }
+            else
+            {
+                backgroundColor = SharedHelpers::FindResource(s_DefaultBackgroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
+            }
+
+            return std::make_tuple(foregroundColor, backgroundColor);
+        }();
+       
+        
+        player.Source(winrt::make<ProgressRingLoading>(StrokeThickness(), size, foregroundColor, backgroundColor));
```

Comment:
This will cause the Composition tree to be rebuilt and the animation to restart every time ApplyLottieAnimation is called.  That's going to be jarring for the user, and it's inefficient. Instead, the properties that you need to change should be stored in the CompositionPropertySet at the root of the IAnimatedVisual and they should be bound to the properties in the IAnimatedVisual via expressions. LottieGen will wire this up for you for colors. If we need this for strokes then we'll enable that in LottieGen too.